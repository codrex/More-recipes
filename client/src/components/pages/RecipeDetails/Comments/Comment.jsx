import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Animation from '../../../common/Animation';
import getInitials from '../../../../utils/getInitials';

/**
 * Comment component
 * @param {object} props
 * @return {React} react element
 */
const Comment = props => (
  <Animation>
    <div className="media-object comment">
      <div className="avatar avatar-sm">{getInitials(props.fullname)}</div>
      <div className="media-body">
        <div className="d-flex justify-content-between align-items-center">
          <h4 className="no-margin">
            {`@${props.username}`}
          </h4>
          <span className="text-center bold time">
            {moment(props.createdAt).fromNow()}
          </span>
        </div>
        <p className="media-text no-margin">
          {props.review}
        </p>
      </div>
    </div>
  </Animation>
);

Comment.propTypes = {
  review: PropTypes.string.isRequired,
  fullname: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired
};

export default Comment;

