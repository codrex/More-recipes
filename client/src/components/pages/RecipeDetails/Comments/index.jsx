import React from 'react';
import PropTypes from 'prop-types';
import NotFound from '../../../common/NotFound';
import Comment from './Comment';
import Loader from '../../../common/Loader';

/**
 * Comments component
 * @param {object} props
 * @return {React} react element
 */
const Comments = (props) => {
  const { comments, isLoading } = props;
  return (
    <div className="col-12 " style={{ minHeight: '63vh' }}>
      {comments.map(({
        id,
        reviewer,
        review,
        createdAt
      }, i) => (
        <Comment
          key={id}
          review={review}
          createdAt={createdAt}
          fullname={reviewer.fullname}
          username={reviewer.username}
          index={i}
        />
      )
      )}
      {comments.length === 0 && !isLoading &&
        <NotFound message="no review found" />
      }
      { isLoading && <Loader loading={isLoading} />
      }
    </div>
  );
};

Comments.defaultProps = {
  comments: [],
  createdAt: '',
  isLoading: true
};
Comments.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape),
  createdAt: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Comments;
