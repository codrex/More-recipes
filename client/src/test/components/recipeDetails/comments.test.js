import React from 'react';
import toJson from 'enzyme-to-json';
import Comments from '../../../components/pages/RecipeDetails/Comments';

const props = {
  comments: [{
    id: 50,
    review: 'cool recipe',
    ReviewerId: 1,
    RecipeId: 1,
    createdAt: '',
    reviewer: {
      id: 1,
      username: 'testi',
      fullname: 'test tester ',
      profilePix: 'UNKNOWN',
      email: 'test.test@gmail.com',
    }
  }],
};

describe('Comments display component ', () => {
  test('expected to match snapshot', () => {
    const component = mount(<Comments {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
});
