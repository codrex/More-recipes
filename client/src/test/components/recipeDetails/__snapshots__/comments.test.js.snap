// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments display component  expected to match snapshot 1`] = `
<Comments
  comments={
    Array [
      Object {
        "RecipeId": 1,
        "ReviewerId": 1,
        "createdAt": "",
        "id": 50,
        "review": "cool recipe",
        "reviewer": Object {
          "email": "test.test@gmail.com",
          "fullname": "test user ",
          "id": 1,
          "profilePix": "UNKNOWN",
          "username": "testuser",
        },
      },
    ]
  }
  createdAt=""
  isLoading={true}
>
  <div
    className="col-12 "
    style={
      Object {
        "minHeight": "63vh",
      }
    }
  >
    <Comment
      createdAt=""
      fullname="test user "
      index={0}
      key="50"
      review="cool recipe"
      username="testuser"
    >
      <Animation
        appear={true}
        name="fade"
        timeOut={600}
      >
        <CSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnter={true}
          transitionEnterTimeout={600}
          transitionLeave={true}
          transitionLeaveTimeout={600}
          transitionName="fade"
        >
          <TransitionGroup
            childFactory={[Function]}
            component="span"
            transitionAppear={true}
            transitionAppearTimeout={600}
            transitionEnter={true}
            transitionEnterTimeout={600}
            transitionLeave={true}
            transitionLeaveTimeout={600}
            transitionName="fade"
          >
            <span>
              <CSSTransitionGroupChild
                appear={true}
                appearTimeout={600}
                enter={true}
                enterTimeout={600}
                key=".0"
                leave={true}
                leaveTimeout={600}
                name="fade"
              >
                <div
                  className="media-object comment"
                  key=".0"
                >
                  <div
                    className="avatar avatar-sm"
                  >
                    TU
                  </div>
                  <div
                    className="media-body"
                  >
                    <div
                      className="d-flex justify-content-between align-items-center"
                    >
                      <h4
                        className="no-margin"
                      >
                        @testuser
                      </h4>
                      <span
                        className="text-center bold time"
                      >
                        Invalid date
                      </span>
                    </div>
                    <p
                      className="media-text no-margin"
                    >
                      cool recipe
                    </p>
                  </div>
                </div>
              </CSSTransitionGroupChild>
            </span>
          </TransitionGroup>
        </CSSTransitionGroup>
      </Animation>
    </Comment>
    <Loader
      loading={true}
    >
      <div
        className="loader"
      >
        <onlyUpdateForKeys(Loader)
          color="#7B4E28"
          loading={true}
          size={60}
        >
          <Loader
            color="#7B4E28"
            loading={true}
            size={60}
          >
            <div
              className="css-igcz5d"
            />
          </Loader>
        </onlyUpdateForKeys(Loader)>
      </div>
    </Loader>
  </div>
</Comments>
`;
